### PR TITLE
opae: update 1.4.0  release notes

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -49,30 +49,32 @@ Major Changes from 1.2.0 to 1.3.0
...
-- Various Static code scan bug fixes
-- Added new FPGA MMIO API to write 512 bits
+- Added support to FPGA Linux kernel Device Feature List (DFL) driver patch set2.
+- Added test cases and Increased test coverage
+- Various bug fixes
+- Various memory leak fixes
+- Various Static code scan bug fixes
+- Added new FPGA MMIO API to write 512 bits
+
+- Operating System: Tested on RedHat 7.6, CentOS7.6  with Linux Kernel 3.10 and 4.19
```
Comment (jelonanX) [inline RELEASE_NOTES.md]: CentOS 7.6 with a space in between?

Comment (anandaravuri) [reply RELEASE_NOTES.md]: Fixed

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,131 +1,128 @@
...
+OPAE is under active development to extend to more hardware platforms, as well as to build up the software stack with additional abstractions to enable more software developers.
+
+OPAE SDK is a collection of libraries and tools to facilitate the development of software applications and accelerators using OPAE. It provides a library implementing the OPAE C API for presenting a streamlined and easy-to-use interface for software applications to discover, access, and manage FPGA devices and accelerators using the OPAE software stack. The OPAE SDK also includes the AFU Simulation Environment (ASE) for end-to-end simulation of accelerator RTL together with software applications using the OPAE C API.
+
+OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplify the development and deployment of FPGA applications, so we explicitly welcome discussions and contributions! The OPAE SDK source, unless otherwise noted, is released under a BSD 3-clause license.
+
+More information about OPAE can be found
+at http://01.org/OPAE.
+
+Open Programmable Acceleration Engine (OPAE) 1.3.0 Release Notes
```
Comment (tswhison) [inline RELEASE_NOTES.md]: 1.4.0

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,131 +1,128 @@
...
+-----------------------------------------------------------------
+
+This document provides the Release Notes for the Open Programmable
+Acceleration Engine (OPAE) 1.3.0 release.
```
Comment (tswhison) [inline RELEASE_NOTES.md]: 1.4.0

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -1,131 +1,128 @@
...
+
+-   Hardware: tightly coupled FPGA products and programmable FPGA
+    acceleration cards for Intel(R) Xeon(R) processors:
+    - Intel(R) PAC with Arria(R) 10 GX FPGA (PCI ID: 0x09c4) FIM version 1.0.3 (1.0 Production)
+    - Intel(R) Xeon with Integrated FPGA (PCI ID: 0xbcc0) FIM version 6.4.0
+
+-   Operating System: tested on Red Hat Enterprise Linux 7.3 and 7.4, Ubuntu 16.04, 
+    SUSE SLE 12 SP3 and CentOS 7.4, with Linux kernels 3.10 through 4.7
+
+Major Changes from 1.2.0 to 1.3.0
```
Comment (tswhison) [inline RELEASE_NOTES.md]: Move this section down such that "Changes from 1.3.0 to 1.4.0" appears here first.

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,131 +1,128 @@
...
+- Various bug fixes
+- Various compiler warning fixes
+- Various memory leak fixes
+
+Major Changes from 1.3.0 to 1.4.0
+----------------------------------
+- Added support to FPGA Linux kernel Device Feature List (DFL) driver patch set2.
+- Added test cases and Increased test coverage
+- Various bug fixes
+- Various memory leak fixes
```
Comment (tswhison) [inline RELEASE_NOTES.md]: Do we need to call this out explicitly?  The previous point "Various bug fixes" would account for memory leaks and static code scan bugs.

File: RELEASE_NOTES.md [hunk 6]
```diff
@@ -17,22 +17,43 @@ OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplif
...
+- Various Static code scan bug fixes
+- Added new FPGA MMIO API to write 512 bits
+
+- Operating System: Tested on RedHat 7.6, CentOS7.6  with Linux Kernel 3.10 and 4.19
```
Comment (jelonanX) [inline RELEASE_NOTES.md]: CentOS 7.6

File: RELEASE_NOTES.md (hunk 2 above)
Comment (anandaravuri) [reply RELEASE_NOTES.md]: fixed

File: RELEASE_NOTES.md (hunk 3 above)
Comment (anandaravuri) [reply RELEASE_NOTES.md]: fixed

File: RELEASE_NOTES.md (hunk 6 above)
Comment (anandaravuri) [reply RELEASE_NOTES.md]: Fixed

**Comment by rlee5-intel** [comment]
I do not believe this is what we have tested.

We should specify OS and Kernel that we have tested.  Also I think this "through" word is very miss-leading.  We should not use this word.

 SUSE SLE 12 SP3 and CentOS 7.4, with Linux kernels 3.10 through 4.19

File: RELEASE_NOTES.md (hunk 5 above)
Comment (rlee5-intel) [reply RELEASE_NOTES.md]: If we want to list these bug fixes/memory leak fixes, can we list bugs that we have fixed in this release?

**Comment by rweight** [comment]
> SUSE SLE 12 SP3 and CentOS 7.4, with Linux kernels 3.10 through 4.19

This is a good point. Maybe this should say "SUSE SLE 12 SP3, CentOS 7.4, and the 4.19 LTS kernel"

File: RELEASE_NOTES.md [hunk 7]
```diff
@@ -17,22 +17,43 @@ OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplif
...
+- Various Static code scan bug fixes
+- Added new FPGA MMIO API to write 512 bits
+
+- Operating System: Tested on RedHat 7.6, CentOS 7.6  with Linux Kernel 3.10 and 4.19
+
+- OPAE & Intel FPGA driver are tested on Intel Programmable Acceleration Card Arria 10 GX FPGA & Intel PAC N3000.
+
+- OPAE & DFL FPGA driver are tested on Intel Programmable Acceleration Card Arria 10 GX FPGA.
+
+- FPGA DFL Linux driver source code patchset2 available
```
Comment (r-rojo) [inline RELEASE_NOTES.md]: Can we also explicitly call out kernel version here (as you did below)?

File: RELEASE_NOTES.md [hunk 8]
```diff
@@ -17,22 +17,43 @@ OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplif
...
 -   Hardware: tightly coupled FPGA products and programmable FPGA
     acceleration cards for Intel(R) Xeon(R) processors:
-    - Intel(R) PAC with Arria(R) 10 GX FPGA (PCI ID: 0x09c4) FIM version 1.0.3 (1.0 Production)
+    - Intel(R) PAC with Arria(R) 10 GX FPGA (PCI ID: 0x09c4) FIM version 1.1.2-1 (1.2 Production)
     - Intel(R) Xeon with Integrated FPGA (PCI ID: 0xbcc0) FIM version 6.4.0
+    - Intel® FPGA Programmable Acceleration Card N3000 (PCI ID: 0x0b30) FIM version D.1.0.13 (1.0 Production)
 
 -   Operating System: tested on Red Hat Enterprise Linux 7.3 and 7.4, Ubuntu 16.04, 
-    SUSE SLE 12 SP3 and CentOS 7.4, with Linux kernels 3.10 through 4.7
+    SUSE SLE 12 SP3 and CentOS 7.4, with Linux kernels 3.10 through 4.19
```
Comment (rweight) [inline RELEASE_NOTES.md]: Should SuSE be listed (I don't know)
Listing the RedHat kernel but not the SuSE kernel seems confusing to me.
And this should be CentOS 7.6 (not 7.4), right?
So maybe something like: SUSE SLE 12 SP3 and CentOS 7.4 (with both the CentOS 7.4 kernel and the community 4.19 LTS kernel)

File: RELEASE_NOTES.md [hunk 9]
```diff
@@ -17,115 +17,62 @@ OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplif
...
+--------------------
 
--   Virtual Function (VF) may fail to attach or detach when using the
-    Linux Red Hat 3.10 kernel
-    
-    This is a known issue with qemu/kvm and libvirt. Refer to the Red
-    Hat website for more information about this issue.
+- In addition to supporting the OPAE driver bundled with OPAE SDK releases, the OPAE SDK libraries now
+support the FPGA driver that has been upstreamed to the Linux Kernel 5.4.
+For more details on this, please see the [OPAE documentation](https://opae.github.io/1.4.0/docs/fpga_dfl_drv/fpga_dfl_drv.html) related to this.
```
Comment (tswhison) [inline RELEASE_NOTES.md]: Getting 404 Not Found for this link.

Comment (anandaravuri) [reply RELEASE_NOTES.md]: we have run doc gen script to generate 1.4.0 folder.

Comment (tswhison) [reply RELEASE_NOTES.md]: Oh, probably because 1.4.0 doesn't exist yet.  Nevermind.

File: RELEASE_NOTES.md [hunk 10]
```diff
@@ -17,115 +17,62 @@ OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplif
...
 
--   The Intel FPGA Dynamic Profiler Tool for OpenCL GUI reports
-    frequency and bandwidth incorrectly
+- DFL FPGA driver pactset2 doesn’t support all the features supported by Intel FPAG driver.
```
Comment (ibarraza) [inline RELEASE_NOTES.md]: patchset2

Comment (anandaravuri) [reply RELEASE_NOTES.md]: fixed

File: RELEASE_NOTES.md [hunk 11]
```diff
@@ -17,115 +17,62 @@ OPAE\'s goal is to accelerate FPGA adoption. It is a community effort to simplif
...
-    
-    This is a known issue with qemu/kvm and libvirt. Refer to the Red
-    Hat website for more information about this issue.
+- In addition to supporting the OPAE driver bundled with OPAE SDK releases, the OPAE SDK libraries now
+support the FPGA driver that has been upstreamed to the Linux Kernel 5.4.
+For more details on this, please see the [OPAE documentation](https://opae.github.io/1.4.0/docs/fpga_dfl_drv/fpga_dfl_drv.html) related to this.
 
--   The Intel FPGA Dynamic Profiler Tool for OpenCL GUI reports
-    frequency and bandwidth incorrectly
+- DFL FPGA driver patchset2 doesn’t support all the features supported by Intel FPAG driver.
```
Comment (tswhison) [inline RELEASE_NOTES.md]: s/FPAG/FPGA/

**Pull request merged**